lib/ddtrace/contrib/rails/core_extensions.rb

```diff
@@ -60,8 +54,8 @@ def render_template_with_datadog(*args)
                        else
                          layout_name.try(:[], 'virtual_path')
                        end
-              @tracing_context[:template_name] = template_name
-              @tracing_context[:layout] = layout
+              active_datadog_span.set_tag('rails.template_name', template_name) if template_name
+              active_datadog_span.set_tag('rails.layout', layout) if layout
```

Comment:
`freeze`ing this string might also help a bit with memory use.

Reply:
Same as above.